### pandas

In [1]:
import pandas as pd

database = [['pepito','pancho','felipe','ignacio','tolomeo'],['M','M','M','M','M'],[132,564,84,864,684]]
database

df = pd.DataFrame(database, columns=['1', '2', '3', '4', '5'], index=['Nombre','Sexo','numero'])

df.style.set_properties(**{'text-align': 'center'})

df

1       2       3        4        5
Nombre  pepito  pancho  felipe  ignacio  tolomeo
Sexo         M       M       M        M        M
numero     132     564      84      864      684

## Importar un dataset IRIS
## Leemos el .csv con pd.read_csv("dirección_del_archivo") y lo guardamos en df_iris

In [3]:
# Read the CSV file
df_iris = pd.read_csv("../databaseiris/iris.csv")
df_iris

sepal.length  sepal.width  petal.length  petal.width    variety
0             5.1          3.5           1.4          0.2     Setosa
1             4.9          3.0           1.4          0.2     Setosa
2             4.7          3.2           1.3          0.2     Setosa
3             4.6          3.1           1.5          0.2     Setosa
4             5.0          3.6           1.4          0.2     Setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  Virginica
146           6.3          2.5           5.0          1.9  Virginica
147           6.5          3.0           5.2          2.0  Virginica
148           6.2          3.4           5.4          2.3  Virginica
149           5.9          3.0           5.1          1.8  Virginica

[150 rows x 5 columns]

## value_counts. 

In [5]:
df_iris.value_counts()

sepal.length  sepal.width  petal.length  petal.width  variety  
5.8           2.7          5.1           1.9          Virginica    2
4.4           2.9          1.4           0.2          Setosa       1
              3.0          1.3           0.2          Setosa       1
              3.2          1.3           0.2          Setosa       1
4.3           3.0          1.1           0.1          Setosa       1
                                                                  ..
7.7           2.6          6.9           2.3          Virginica    1
              2.8          6.7           2.0          Virginica    1
              3.0          6.1           2.3          Virginica    1
              3.8          6.7           2.2          Virginica    1
7.9           3.8          6.4           2.0          Virginica    1
Name: count, Length: 149, dtype: int64

## Podemos usar len(df_iris) para contar el numero de filas en la DataFrame, en este caso 150 filas

In [6]:
len(df_iris)

150

## df_iris.shape nos indica el un par ordenado donde el primer valor indica el numero de filas y 5 el numero de columnas.

In [7]:
df_iris.shape

(150, 5)

## df_iris.nunique() nos indica el numero de valores unicos de cada columna

In [8]:
df_iris.nunique()

sepal.length    35
sepal.width     23
petal.length    43
petal.width     22
variety          3
dtype: int64

## df_iris.describe() nos muestra por columnas el "valores", "valor medio", "std", "valor minimo", "25%" ,"50%" ,"75%" , "valor maximo".

In [16]:
df_iris.describe()

sepal.length  sepal.width  petal.length  petal.width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

## df_iris.sum() realizar la suma de todos los datos de una variable/columna; es curioso notar que concatena las variables de texto

In [39]:
df_iris.sum()

sepal.length                                                876.5
sepal.width                                                 458.6
petal.length                                                563.7
petal.width                                                 179.9
variety         SetosaSetosaSetosaSetosaSetosaSetosaSetosaSeto...
dtype: object

## df_iris.count() cuenta los datos de una variable, en este caso todas tienen 150 datos (es decir, filas)

In [40]:
df_iris.count()

sepal.length    150
sepal.width     150
petal.length    150
petal.width     150
variety         150
dtype: int64

## df_iris.median() tiene de salida el valor de medio para cada valor; en este caso fallará por la columna variety con str (texto), por ello se debe indicar las columnas donde se aplicará median()

In [42]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].median()

sepal.length    5.80
sepal.width     3.00
petal.length    4.35
petal.width     1.30
dtype: float64

## df_iris.quantile() tiene de salia los cuantiles de cada objeto

In [43]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].quantile([0.25,0.75])

sepal.length  sepal.width  petal.length  petal.width
0.25           5.1          2.8           1.6          0.3
0.75           6.4          3.3           5.1          1.8

## df_iris.apply() : comenta la documentación que se aplica una función en cada objeto.

In [ ]:
df_iris.apply()

## df_iris.min() da de salida el valor minimo de cada variable

In [44]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].min()

sepal.length    4.3
sepal.width     2.0
petal.length    1.0
petal.width     0.1
dtype: float64

## df_iris.max() imprime el valor maximo para cada variable

In [45]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].max()

sepal.length    7.9
sepal.width     4.4
petal.length    6.9
petal.width     2.5
dtype: float64

## df_iris.mean() da el valor medio de cada variable

In [47]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].mean()

sepal.length    5.843333
sepal.width     3.057333
petal.length    3.758000
petal.width     1.199333
dtype: float64

## df_iris.var() da la varianza de cada variable

In [48]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].var()

sepal.length    0.685694
sepal.width     0.189979
petal.length    3.116278
petal.width     0.581006
dtype: float64

## df_iris.std() da la desviación estandar de cada varible

In [49]:
df_iris[['sepal.length','sepal.width','petal.length','petal.width']].std()

sepal.length    0.828066
sepal.width     0.435866
petal.length    1.765298
petal.width     0.762238
dtype: float64